### ISSUE TITLE
'pycontrol._keep_python_running' incorrectly finalizes interpreter even if 'ParticleInjector' uses a python profile.

### ISSUE THREAD

**Issue by ekapadi** [SmileiPIC/Smilei#216]
## Description

Define a  'ParticleInjector' using a user-defined python profile.

```
Watch this SIGSEGV when smilei tries to run a python function on a finalized interpreter.  :(
```

## Steps to reproduce the problem
See previous.
## Parameters

A pull request for the fix to this is available.  Please contact me directly.

**Comment by mccoys** [comment]
You are welcome to submit a pull request

Thank you for spotting this bug

**Comment by mccoys** [comment]
@ekapadi Will you be able to submit a pull request ?

**Comment by mccoys** [comment]
this should be fixed in v4.4
